Fix units to be able to compose from equivalent units that have prefixes

fastakash opened the pull request:
Fix #4104.

Made changes as suggested and added some tests also.  My first commit to astropy and github!  Took me a long time to figure out  all the steps.

fastakash commented:
Hi @bsipocz   can you please add the label work-in-progress?  I initially thought my work was fine but now see that a failing unit-test is because of the change I made.  I will investigate further.

Sorry for the PR being premature

fastakash commented:
The label "work in progress" may now be removed.  Hoping for a review/comment next!

bsipocz commented:
@fastakash - I've changed the title of your PR to be more informative about its content (as probably none of us knows the issues by their number).

mhvk commented on astropy/units/tests/test_units.py:
```diff
@@ -128,6 +128,18 @@ def test_units_manipulation():
 
 
+def test_compose():
+    x =  u.m.compose(units=(u.m))
```
I think this is sufficiently tested elsewhere, so I would instead replace it with something else that uses a tuple (but then add the trailing `,`!). Also, maybe make the test a bit more realistic by adding multiple possible units. E.g.,
```
x = (u.km/u.s).compose(units=(u.pc, u.Myr))
assert x[0].bases == [u.pc, u.Myr]
assert_allclose(x[0].scale, 1.0227121650537077)

mhvk commented on astropy/units/tests/test_units.py:
```diff
@@ -128,6 +128,18 @@ def test_units_manipulation():
+    x = u.m.compose(units=[u.km])
+    assert x[0].bases[0] is u.km
+    assert x[0].scale == 0.001
+
```
Of course, also important to test whether things *fail* when `include_prefix_units=False`:
```
with raises(u.UnitsError):
    (u.km/u.s).compose(units=(u.pc, u.Myr), include_prefix_units=False)
```

mhvk commented on CHANGES.rst:
```diff
@@ -135,6 +135,9 @@ astropy.time
 ^^^^^^^^^^^^^
 
+- When a prefix unit is specified, include_prefix_units should be automatic
+  [#4104, #6957]
```
We usually just specify the PR, not the original issue.

fastakash commented:
@mhvk Thanks for patiently explaining everything to me!  I hope I addressed all your feedback.  I will be more careful with reviewing all the existing unittests in the future.  I know this was a small task that you could have probably finished yourself in minutes!  I hope to pick up more and bigger tasks, especially in the summer.

fastakash commented:
Hi @mhvk   Thanks again for your help.  Do you think my updates are ok?  If you are busy, please ignore this message!  I will start looking at other issues.. thanks

mhvk commented on astropy/units/core.py:
```diff
@@ -1140,6 +1140,8 @@ def compose(self, equivalencies=[], units=None, max_depth=2,
         include_prefix_units : bool, optional
+            Default is `True` if a sequence is passed in to ``units``,
+            `False` otherwise.
             When `True`, include prefixed units in the result.
```
THis got a bit mangled. Could you change it to
```
When `True`, include prefixed units in the result.
Default is `True` if a sequence is passed in to ``units``,
`False` otherwise.
```

mhvk commented:
@fastakash - apart from the very minor comment about the docstring, this all looks good. If you change that one, add `[ci skip]` to your commit message, since it is not necessary to run the tests again.

mhvk commented:
@fastakash - I realized the changelog was for 3.0, but this will have to be in 3.1, so I went ahead and moved it (rebasing to be sure I picked up the new changelong entries). Otherwise, it all looks good, so I'll merge. Thanks!
